src/main/java/Student.java

```diff
@@ -0,0 +1,146 @@
+		int remainingHours = curriculum.getTotalDurationHrs(); 
+        final int hoursPerDay = 8; 
+        LocalDateTime endDateTime = startDateTime;
+
+        while (remainingHours > 0) {
+            // Check if it's a work day 
+            if (endDateTime.getDayOfWeek() != DayOfWeek.SATURDAY && endDateTime.getDayOfWeek() != DayOfWeek.SUNDAY) {
+                // Check if it's within working hours
+                if (endDateTime.getHour() >= 10 && endDateTime.getHour() < 18) {
+                    int hoursToAdd = Math.min(hoursPerDay, remainingHours); // Add 8 hours or remaining hours, whichever is smaller
+                    endDateTime = endDateTime.plusHours(hoursToAdd);
+                    remainingHours -= hoursToAdd;
+                } else {
+                    // Move to the next day at 10 am 
+                	endDateTime = endDateTime.plusDays(1).withHour(10).withMinute(0);
+                }
+            } else {
+                // Move to the next day at 10 am
+            	endDateTime = endDateTime.plusDays(1).withHour(10).withMinute(0);
+            }
+        }
+
+        return endDateTime;
+    }
+	
+    
+        public void getProgress(LocalDateTime startDateTime, Curriculum curriculum) {
+        	LocalDateTime currentDateTime = LocalDateTime.now();
+        	LocalDateTime endDateTime = getEndDateTime(startDateTime, curriculum); 
+        	
+        	int diff1 = currentDateTime.compareTo(startDateTime);
+        	int diff2 = currentDateTime.compareTo(endDateTime); 
+        	
+        	if (diff1 >= 0) { //training has been started
+                if (diff2 < 0) { //training not completed
+                    printProgressNotFinished(startDateTime, curriculum);
+                } else { //training is completed
+                    printProgressCompleted(startDateTime, curriculum);
+                }
+            } else {
+                printProgressBeforeStart(startDateTime, curriculum);
+            }
+        }
+        
+        
+        //internal helper counts of duration between to dates
+        public long calculateTotalHours(LocalDateTime startTimePoint, LocalDateTime endTimePoint) {
+            long totalHours = Duration.between(startTimePoint, endTimePoint).toHours();
+            long minutes = Duration.between(startTimePoint, endTimePoint).toMinutesPart();
+            
+            if (minutes >= 30) {
+                totalHours++;
+            }
+            return totalHours;
+        }
+        
+        //internal helper to print amount of time in appropriate format
+        void printProgressMessage(long totalHours, String description) {
+            if (totalHours / 24 != 0) {
+                System.out.printf("%d d ", totalHours / 24);
+            }
+            if (totalHours % 24 != 0) {
+                System.out.printf("%d h ", totalHours % 24);
+            }
+            System.out.printf("%s.%n", description);
+        }
```

Comment:
if it's 'internal helper', it should probably return the value instead of printing it. Responsibility to print the final report should be kept in once place